### PR TITLE
Dmp 3293: delete transcription and annotation from unstructured

### PR DESCRIPTION
**Pull request by mestebanez**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-3293

### Change description ###

Add a new task to remove arm resource from unstructured based on last modified date exceeding the configured number of weeks (30 by default)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            ChronoUnit.WEEKS
+        );
+
+        List<Integer> recordsMarkedForDeletion
+            = externalObjectDirectoryRepository
+            .findMediaFileIdsIn2StorageLocationsBeforeTime(pageable,
+                                                           EodHelper.storedStatus(),
+                                                           EodHelper.storedStatus(),
+                                                           EodHelper.unstructuredLocation(),
+                                                           EodHelper.armLocation(),
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java:69]: DMP-3293 suggests the ARM check should consider `last_modified_ts is 24hrs or more ago (darts.data-management.retention-period.inbound.arm-minimum)`, whereas I think this is considering the 30 weeks?

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                                                           lastModifiedBefore);
+
+        log.debug("Identified records to be marked for deletion  {}",  recordsMarkedForDeletion.stream().map(Object::toString));
+
+        UserAccountEntity user = userAccountRepository.findSystemUser(systemUserHelper.findSystemUserGuid("housekeeping"));
+        eodHelper.updateStatus(
+            EodHelper.markForDeletionStatus(),
+            user,
+            recordsMarkedForDeletion,
+            OffsetDateTime.now()
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java:79]: Consider `currentTimeHelper.currentOffsetDateTime()` for consistency and improved testability.

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/InboundAnnotationTranscriptionDeleterProcessorImpl.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                                           lastModifiedBefore);
+
+        log.debug("Identified records to be marked for deletion  {}",  recordsMarkedForDeletion.stream().map(Object::toString));
+
+        UserAccountEntity user = userAccountRepository.findSystemUser(systemUserHelper.findSystemUserGuid("housekeeping"));
+        eodHelper.updateStatus(
+            EodHelper.markForDeletionStatus(),
+            user,
+            recordsMarkedForDeletion,
+            OffsetDateTime.now()
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/InboundAnnotationTranscriptionDeleterProcessorImpl.java:68]: Consider currentTimeHelper.currentOffsetDateTime() for consistency and improved testability.

**Comment by chrisbellingham-hmcts** [comment]
Looks like this also includes DMP-3292, is that intended?

File: src/main/java/uk/gov/hmcts/darts/task/service/impl/AutomatedTaskServiceImpl.java [hunk 4]
```diff
@@ -872,6 +908,40 @@ private void rescheduleGenerateCaseDocumentAutomatedTask() {
...
+            );
+            Trigger trigger = createAutomatedTaskTrigger(generateCaseDocumentAutomatedTask);
+            taskScheduler.schedule(generateCaseDocumentAutomatedTask, trigger);
+        } else {
+            taskScheduler.schedule(triggerAndAutomatedTask.getAutomatedTask(), triggerAndAutomatedTask.getTrigger());
+        }
+    }
+
+    private void rescheduleUnstructuredTranscriptionAndAnnotationDeleterAutomatedTask() {
+        TriggerAndAutomatedTask triggerAndAutomatedTask = getTriggerAndAutomatedTask(INBOUND_TRANSCRIPTION_ANNOTATION_DELETER_TASK_NAME.getTaskName());
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/task/service/impl/AutomatedTaskServiceImpl.java:929]: good shout. Corrected. I have also added some additional tests to catch this misconfiguration.

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java (hunk 1 above)
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/UnstructuredAnnotationTranscriptionDeleterProcessorImpl.java:65]: So the check takes 2 dates. The first one is the 30 weeks for unstructured and the second one is 24 hours for arm

**Comment by mestebanez** [comment]
> Looks like this also includes DMP-3292, is that intended?

yep. This PR builds on DMP-3292 so should be addressed first.

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**